server/sonar-server/src/main/java/org/sonar/server/qualityprofile/RegisterQualityProfiles.java

```diff
@@ -86,11 +111,80 @@ private void registerPerQualityProfile(DbSession session, DefinedQProfile qualit
       qualityProfile.getLoadedTemplateType(), PROCESSED_ORGANIZATIONS_BATCH_SIZE);
   }
 
-  private void registerPerQualityProfileAndOrganization(DbSession session, DefinedQProfile qualityProfile, OrganizationDto organization, List<ActiveRuleChange> changes) {
-    LOGGER.debug("Register profile {} for organization {}", qualityProfile.getQProfileName(), organization.getKey());
+  private void registerPerQualityProfileAndOrganization(DbSession session,
+    DefinedQProfile definedQProfile, OrganizationDto organization, List<ActiveRuleChange> changes, Profiler profiler) {
+    profiler.start();
+
+    definedQProfileInsert.create(session, definedQProfile, organization, changes);
 
-    definedQProfileCreation.create(session, qualityProfile, organization, changes);
     session.commit();
+
+    profiler.stopDebug(format("Register profile %s for organization %s", definedQProfile.getQProfileName(), organization.getKey()));
+  }
+
+  @VisibleForTesting
+  static class SortByParentName implements Comparator<DefinedQProfile> {
+    private final Map<String, DefinedQProfile> buildersByName;
+    @VisibleForTesting
+    final Map<String, Integer> depthByBuilder;
+
+    @VisibleForTesting
+    SortByParentName(Collection<DefinedQProfile> builders) {
+      buildersByName = builders.stream()
+        .collect(MoreCollectors.uniqueIndex(DefinedQProfile::getName, Function.identity(), builders.size()));
+      Map<String, Integer> depthByBuilder = new HashMap<>();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Rename "depthByBuilder" which hides the field declared at line 128. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)